Minor Docker changes and fixing yarn in Dockerfile

zoidyzoidzoid opened the pull request.

zoidyzoidzoid committed:
Tidy up Dockerfile

zoidyzoidzoid committed:
We need to update apt-get to fetch the new indices

zoidyzoidzoid committed:
Also install newer Node than Debian 9 has

zoidyzoidzoid committed:
Make Docker build context smaller

Sending build context to Docker daemon  113.4MB

Sending build context to Docker daemon  479.7kB

sewagodimo commented on Dockerfile:
```diff
@@ -7,21 +7,27 @@ ENV PYTHONUNBUFFERED 1
-	pipenv install --system --deploy &&\
-	pipenv run django-admin migrate
- EXPOSE 8000
+RUN set -ex; \
```
what does this do? :eyes:

zoidyzoidzoid replied:
Sorry, I should've explained in the commit message.

So it sets two shell options.

`set -e` means that if one of the following commands exits with an error (non-zero exit code), then it'll exit the same error code, instead of continuing on. A cleaner way to do the same `&&` spaghetti, that I borrowed from Praekelt's pretty Dockerfiles [here](https://github.com/praekeltfoundation/docker-django-bootstrap/blob/develop/Dockerfile)

`set -x` turns on shell tracing, which means that it prints each command it runs too, which means we get a prettier docker build output too.

Example output:

<details>
Sending build context to Docker daemon  481.3kB
Step 1/9 : FROM python:3.6
 ---> 4f13b7f2138e
Step 2/9 : ENV PYTHONUNBUFFERED 1
 ---> Using cache
 ---> 2b6d482802a0
Step 3/9 : ENV IN_DOCKER 1
 ---> Running in f1746a6a8663
Removing intermediate container f1746a6a8663
 ---> 9415c213570f
Step 4/9 : RUN mkdir /buza-website
 ---> Running in 2f9cf3e7c43c
Removing intermediate container 2f9cf3e7c43c
 ---> 22d03a369c47
Step 5/9 : RUN set -ex;   apt-get update;   apt-get install apt-transport-https;   curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -;   echo "deb https://dl.yarnpkg.com/debian/ stable main" >/etc/apt/sources.list.d/yarn.list;   curl -sL https://deb.nodesource.com/setup_10.x -o nodesource_setup.sh;   bash nodesource_setup.sh;   apt-get update;   apt-get install -y python3-pip yarn nodejs
 ---> Running in df6bc6495c1a
+ apt-get update
Ign:1 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Get:2 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease [91.0 kB]
Get:3 http://cdn-fastly.deb.debian.org/debian stretch Release [118 kB]
Get:5 http://cdn-fastly.deb.debian.org/debian stretch Release.gpg [2434 B]
Get:4 http://security-cdn.debian.org/debian-security stretch/updates InRelease [94.3 kB]
Get:6 http://cdn-fastly.deb.debian.org/debian stretch-updates/main amd64 Packages [5152 B]
Get:7 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 Packages [7089 kB]
Get:8 http://security-cdn.debian.org/debian-security stretch/updates/main amd64 Packages [463 kB]
Fetched 7863 kB in 2s (3536 kB/s)
Reading package lists...
+ apt-get install apt-transport-https
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  apt-transport-https
0 upgraded, 1 newly installed, 0 to remove and 59 not upgraded.
Need to get 171 kB of archives.
After this operation, 243 kB of additional disk space will be used.
Get:1 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 apt-transport-https amd64 1.4.8 [171 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 171 kB in 0s (1465 kB/s)
Selecting previously unselected package apt-transport-https.
(Reading database ... 30224 files and directories currently installed.)
Preparing to unpack .../apt-transport-https_1.4.8_amd64.deb ...
Unpacking apt-transport-https (1.4.8) ...
Setting up apt-transport-https (1.4.8) ...
+ curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg
+ apt-key add -
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK
+ echo deb https://dl.yarnpkg.com/debian/ stable main
+ curl -sL https://deb.nodesource.com/setup_10.x -o nodesource_setup.sh
+ bash nodesource_setup.sh

## Installing the NodeSource Node.js 10.x repo...


## Populating apt-get cache...

+ apt-get update
Ign:1 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Hit:2 http://security-cdn.debian.org/debian-security stretch/updates InRelease
Hit:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease
Hit:4 http://cdn-fastly.deb.debian.org/debian stretch Release
Get:6 https://dl.yarnpkg.com/debian stable InRelease [13.3 kB]
Get:7 https://dl.yarnpkg.com/debian stable/main all Packages [8242 B]
Get:8 https://dl.yarnpkg.com/debian stable/main amd64 Packages [8242 B]
Fetched 29.8 kB in 0s (32.4 kB/s)
Reading package lists...

## Installing packages required for setup: lsb-release...

+ apt-get install -y lsb-release > /dev/null 2>&1

## Confirming "stretch" is supported...

+ curl -sLf -o /dev/null 'https://deb.nodesource.com/node_10.x/dists/stretch/Release'

## Adding the NodeSource signing key to your keyring...

+ curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK

## Creating apt sources list file for the NodeSource Node.js 10.x repo...

+ echo 'deb https://deb.nodesource.com/node_10.x stretch main' > /etc/apt/sources.list.d/nodesource.list
+ echo 'deb-src https://deb.nodesource.com/node_10.x stretch main' >> /etc/apt/sources.list.d/nodesource.list

## Running `apt-get update` for you...

+ apt-get update
Hit:1 http://security-cdn.debian.org/debian-security stretch/updates InRelease
Ign:2 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Get:4 https://deb.nodesource.com/node_10.x stretch InRelease [4612 B]
Hit:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease
Hit:5 http://cdn-fastly.deb.debian.org/debian stretch Release
Hit:6 https://dl.yarnpkg.com/debian stable InRelease
Get:7 https://deb.nodesource.com/node_10.x stretch/main amd64 Packages [769 B]
Fetched 5381 B in 0s (6911 B/s)
Reading package lists...

## Run `sudo apt-get install -y nodejs` to install Node.js 10.x and npm
## You may also need development tools to build native addons:
     sudo apt-get install gcc g++ make
## To install the Yarn package manager, run:
     curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
     echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
     sudo apt-get update && sudo apt-get install yarn


+ apt-get update
Ign:1 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Hit:2 http://security-cdn.debian.org/debian-security stretch/updates InRelease
Hit:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease
Hit:4 http://cdn-fastly.deb.debian.org/debian stretch Release
Hit:5 https://dl.yarnpkg.com/debian stable InRelease
Hit:6 https://deb.nodesource.com/node_10.x stretch InRelease
Reading package lists...
+ apt-get install -y python3-pip yarn nodejs
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  build-essential dbus libapparmor1 libdbus-glib-1-2 libpython3-dev
  libpython3.5 libpython3.5-dev libpython3.5-minimal libpython3.5-stdlib
  python-pip-whl python3-cffi-backend python3-crypto python3-cryptography
  python3-dbus python3-dev python3-gi python3-idna python3-keyring
  python3-keyrings.alt python3-pkg-resources python3-pyasn1
  python3-secretstorage python3-setuptools python3-six python3-wheel
  python3-xdg python3.5 python3.5-dev python3.5-minimal
Suggested packages:
  default-dbus-session-bus | dbus-session-bus python3-crypto-dbg
  python-crypto-doc python-cryptography-doc python3-cryptography-vectors
  python-dbus-doc python3-dbus-dbg gnome-keyring libkf5wallet-bin
  gir1.2-gnomekeyring-1.0 python3-pykde4 doc-base python-secretstorage-doc
  python-setuptools-doc python3.5-venv python3.5-doc binfmt-support
The following NEW packages will be installed:
  build-essential dbus libapparmor1 libdbus-glib-1-2 libpython3-dev
  libpython3.5 libpython3.5-dev nodejs python-pip-whl python3-cffi-backend
  python3-crypto python3-cryptography python3-dbus python3-dev python3-gi
  python3-idna python3-keyring python3-keyrings.alt python3-pip
  python3-pkg-resources python3-pyasn1 python3-secretstorage
  python3-setuptools python3-six python3-wheel python3-xdg python3.5-dev yarn
The following packages will be upgraded:
  libpython3.5-minimal libpython3.5-stdlib python3.5 python3.5-minimal
4 upgraded, 28 newly installed, 0 to remove and 55 not upgraded.
Need to get 63.9 MB of archives.
After this operation, 147 MB of additional disk space will be used.
Get:1 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3.5 amd64 3.5.3-1+deb9u1 [229 kB]
Get:2 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3.5-minimal amd64 3.5.3-1+deb9u1 [1691 kB]
Get:8 https://dl.yarnpkg.com/debian stable/main amd64 yarn all 1.12.3-1 [833 kB]
Get:3 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libpython3.5-stdlib amd64 3.5.3-1+deb9u1 [2167 kB]
Get:13 https://deb.nodesource.com/node_10.x stretch/main amd64 nodejs amd64 10.14.2-1nodesource1 [15.1 MB]
Get:4 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libpython3.5-minimal amd64 3.5.3-1+deb9u1 [573 kB]
Get:5 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libapparmor1 amd64 2.11.0-3+deb9u2 [78.9 kB]
Get:6 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 dbus amd64 1.10.26-0+deb9u1 [210 kB]
Get:7 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 build-essential amd64 12.3 [7346 B]
Get:9 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libdbus-glib-1-2 amd64 0.108-2 [206 kB]
Get:10 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libpython3.5 amd64 3.5.3-1+deb9u1 [1372 kB]
Get:11 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libpython3.5-dev amd64 3.5.3-1+deb9u1 [37.7 MB]
Get:12 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libpython3-dev amd64 3.5.3-1 [18.7 kB]
Get:14 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python-pip-whl all 9.0.1-2 [1406 kB]
Get:15 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-cffi-backend amd64 1.9.1-2 [70.1 kB]
Get:16 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-crypto amd64 2.6.1-7 [259 kB]
Get:17 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-idna all 2.2-1 [32.7 kB]
Get:18 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-pyasn1 all 0.1.9-2 [34.5 kB]
Get:19 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-pkg-resources all 33.1.1-1 [137 kB]
Get:20 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-setuptools all 33.1.1-1 [215 kB]
Get:21 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-six all 1.10.0-3 [14.4 kB]
Get:22 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-cryptography amd64 1.7.1-3 [210 kB]
Get:23 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-dbus amd64 1.2.4-1+b1 [184 kB]
Get:24 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3.5-dev amd64 3.5.3-1+deb9u1 [413 kB]
Get:25 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-dev amd64 3.5.3-1 [1154 B]
Get:26 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-gi amd64 3.22.0-2 [473 kB]
Get:27 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-secretstorage all 2.3.1-2 [14.2 kB]
Get:28 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-keyring all 10.1-1 [36.8 kB]
Get:29 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-keyrings.alt all 1.3-1 [16.2 kB]
Get:30 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-pip all 9.0.1-2 [142 kB]
Get:31 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-wheel all 0.29.0-2 [51.8 kB]
Get:32 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 python3-xdg all 0.25-4 [35.7 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 63.9 MB in 6s (10.2 MB/s)
(Reading database ... 30248 files and directories currently installed.)
Preparing to unpack .../00-python3.5_3.5.3-1+deb9u1_amd64.deb ...
Unpacking python3.5 (3.5.3-1+deb9u1) over (3.5.3-1) ...
Preparing to unpack .../01-python3.5-minimal_3.5.3-1+deb9u1_amd64.deb ...
Unpacking python3.5-minimal (3.5.3-1+deb9u1) over (3.5.3-1) ...
Preparing to unpack .../02-libpython3.5-stdlib_3.5.3-1+deb9u1_amd64.deb ...
Unpacking libpython3.5-stdlib:amd64 (3.5.3-1+deb9u1) over (3.5.3-1) ...
Preparing to unpack .../03-libpython3.5-minimal_3.5.3-1+deb9u1_amd64.deb ...
Unpacking libpython3.5-minimal:amd64 (3.5.3-1+deb9u1) over (3.5.3-1) ...
Selecting previously unselected package libapparmor1:amd64.
Preparing to unpack .../04-libapparmor1_2.11.0-3+deb9u2_amd64.deb ...
Unpacking libapparmor1:amd64 (2.11.0-3+deb9u2) ...
Selecting previously unselected package dbus.
Preparing to unpack .../05-dbus_1.10.26-0+deb9u1_amd64.deb ...
Unpacking dbus (1.10.26-0+deb9u1) ...
Selecting previously unselected package build-essential.
Preparing to unpack .../06-build-essential_12.3_amd64.deb ...
Unpacking build-essential (12.3) ...
Selecting previously unselected package libdbus-glib-1-2:amd64.
Preparing to unpack .../07-libdbus-glib-1-2_0.108-2_amd64.deb ...
Unpacking libdbus-glib-1-2:amd64 (0.108-2) ...
Selecting previously unselected package libpython3.5:amd64.
Preparing to unpack .../08-libpython3.5_3.5.3-1+deb9u1_amd64.deb ...
Unpacking libpython3.5:amd64 (3.5.3-1+deb9u1) ...
Selecting previously unselected package libpython3.5-dev:amd64.
Preparing to unpack .../09-libpython3.5-dev_3.5.3-1+deb9u1_amd64.deb ...
Unpacking libpython3.5-dev:amd64 (3.5.3-1+deb9u1) ...
Selecting previously unselected package libpython3-dev:amd64.
Preparing to unpack .../10-libpython3-dev_3.5.3-1_amd64.deb ...
Unpacking libpython3-dev:amd64 (3.5.3-1) ...
Selecting previously unselected package nodejs.
Preparing to unpack .../11-nodejs_10.14.2-1nodesource1_amd64.deb ...
Unpacking nodejs (10.14.2-1nodesource1) ...
Selecting previously unselected package python-pip-whl.
Preparing to unpack .../12-python-pip-whl_9.0.1-2_all.deb ...
Unpacking python-pip-whl (9.0.1-2) ...
Selecting previously unselected package python3-cffi-backend.
Preparing to unpack .../13-python3-cffi-backend_1.9.1-2_amd64.deb ...
Unpacking python3-cffi-backend (1.9.1-2) ...
Selecting previously unselected package python3-crypto.
Preparing to unpack .../14-python3-crypto_2.6.1-7_amd64.deb ...
Unpacking python3-crypto (2.6.1-7) ...
Selecting previously unselected package python3-idna.
Preparing to unpack .../15-python3-idna_2.2-1_all.deb ...
Unpacking python3-idna (2.2-1) ...
Selecting previously unselected package python3-pyasn1.
Preparing to unpack .../16-python3-pyasn1_0.1.9-2_all.deb ...
Unpacking python3-pyasn1 (0.1.9-2) ...
Selecting previously unselected package python3-pkg-resources.
Preparing to unpack .../17-python3-pkg-resources_33.1.1-1_all.deb ...
Unpacking python3-pkg-resources (33.1.1-1) ...
Selecting previously unselected package python3-setuptools.
Preparing to unpack .../18-python3-setuptools_33.1.1-1_all.deb ...
Unpacking python3-setuptools (33.1.1-1) ...
Selecting previously unselected package python3-six.
Preparing to unpack .../19-python3-six_1.10.0-3_all.deb ...
Unpacking python3-six (1.10.0-3) ...
Selecting previously unselected package python3-cryptography.
Preparing to unpack .../20-python3-cryptography_1.7.1-3_amd64.deb ...
Unpacking python3-cryptography (1.7.1-3) ...
Selecting previously unselected package python3-dbus.
Preparing to unpack .../21-python3-dbus_1.2.4-1+b1_amd64.deb ...
Unpacking python3-dbus (1.2.4-1+b1) ...
Selecting previously unselected package python3.5-dev.
Preparing to unpack .../22-python3.5-dev_3.5.3-1+deb9u1_amd64.deb ...
Unpacking python3.5-dev (3.5.3-1+deb9u1) ...
Selecting previously unselected package python3-dev.
Preparing to unpack .../23-python3-dev_3.5.3-1_amd64.deb ...
Unpacking python3-dev (3.5.3-1) ...
Selecting previously unselected package python3-gi.
Preparing to unpack .../24-python3-gi_3.22.0-2_amd64.deb ...
Unpacking python3-gi (3.22.0-2) ...
Selecting previously unselected package python3-secretstorage.
Preparing to unpack .../25-python3-secretstorage_2.3.1-2_all.deb ...
Unpacking python3-secretstorage (2.3.1-2) ...
Selecting previously unselected package python3-keyring.
Preparing to unpack .../26-python3-keyring_10.1-1_all.deb ...
Unpacking python3-keyring (10.1-1) ...
Selecting previously unselected package python3-keyrings.alt.
Preparing to unpack .../27-python3-keyrings.alt_1.3-1_all.deb ...
Unpacking python3-keyrings.alt (1.3-1) ...
Selecting previously unselected package python3-pip.
Preparing to unpack .../28-python3-pip_9.0.1-2_all.deb ...
Unpacking python3-pip (9.0.1-2) ...
Selecting previously unselected package python3-wheel.
Preparing to unpack .../29-python3-wheel_0.29.0-2_all.deb ...
Unpacking python3-wheel (0.29.0-2) ...
Selecting previously unselected package python3-xdg.
Preparing to unpack .../30-python3-xdg_0.25-4_all.deb ...
Unpacking python3-xdg (0.25-4) ...
Selecting previously unselected package yarn.
Preparing to unpack .../31-yarn_1.12.3-1_all.deb ...
Unpacking yarn (1.12.3-1) ...
Setting up python-pip-whl (9.0.1-2) ...
Processing triggers for mime-support (3.60) ...
Setting up python3-cffi-backend (1.9.1-2) ...
Setting up nodejs (10.14.2-1nodesource1) ...
Setting up build-essential (12.3) ...
Setting up python3-crypto (2.6.1-7) ...
Setting up libdbus-glib-1-2:amd64 (0.108-2) ...
Setting up yarn (1.12.3-1) ...
Setting up python3-idna (2.2-1) ...
Setting up python3-xdg (0.25-4) ...
Setting up python3-keyrings.alt (1.3-1) ...
Setting up python3-six (1.10.0-3) ...
Setting up python3-wheel (0.29.0-2) ...
Setting up python3-pkg-resources (33.1.1-1) ...
Setting up python3-gi (3.22.0-2) ...
Setting up python3-pyasn1 (0.1.9-2) ...
Processing triggers for libc-bin (2.24-11+deb9u3) ...
Setting up libapparmor1:amd64 (2.11.0-3+deb9u2) ...
Setting up python3-pip (9.0.1-2) ...
Setting up dbus (1.10.26-0+deb9u1) ...
invoke-rc.d: could not determine current runlevel
invoke-rc.d: policy-rc.d denied execution of start.
Setting up python3-setuptools (33.1.1-1) ...
Setting up libpython3.5-minimal:amd64 (3.5.3-1+deb9u1) ...
Setting up python3-cryptography (1.7.1-3) ...
Setting up python3-dbus (1.2.4-1+b1) ...
Setting up libpython3.5-stdlib:amd64 (3.5.3-1+deb9u1) ...
Setting up python3.5-minimal (3.5.3-1+deb9u1) ...
Setting up python3-secretstorage (2.3.1-2) ...
Setting up python3.5 (3.5.3-1+deb9u1) ...
Setting up python3-keyring (10.1-1) ...
Setting up libpython3.5:amd64 (3.5.3-1+deb9u1) ...
Setting up libpython3.5-dev:amd64 (3.5.3-1+deb9u1) ...
Setting up python3.5-dev (3.5.3-1+deb9u1) ...
Setting up libpython3-dev:amd64 (3.5.3-1) ...
Setting up python3-dev (3.5.3-1) ...
Processing triggers for libc-bin (2.24-11+deb9u3) ...
Removing intermediate container df6bc6495c1a
 ---> 8ddd598cea49
Step 6/9 : WORKDIR /buza-website
 ---> Running in 7d13d5dd3dae
Removing intermediate container 7d13d5dd3dae
 ---> 5fea94903f11
Step 7/9 : COPY . /buza-website
 ---> b9c129f1366e
Step 8/9 : RUN set -ex;   pip install pipenv;   yarn;   cp -p .env.example .env;   pipenv install --system --deploy;   pipenv run django-admin migrate
 ---> Running in da9e5e261e73
+ pip install pipenv
Collecting pipenv
  Downloading https://files.pythonhosted.org/packages/13/b4/3ffa55f77161cff9a5220f162670f7c5eb00df52e00939e203f601b0f579/pipenv-2018.11.26-py3-none-any.whl (5.2MB)
Requirement already satisfied: setuptools>=36.2.1 in /usr/local/lib/python3.6/site-packages (from pipenv) (40.2.0)
Collecting certifi (from pipenv)
  Downloading https://files.pythonhosted.org/packages/9f/e0/accfc1b56b57e9750eba272e24c4dddeac86852c2bebd1236674d7887e8a/certifi-2018.11.29-py2.py3-none-any.whl (154kB)
Collecting virtualenv (from pipenv)
  Downloading https://files.pythonhosted.org/packages/7c/17/9b7b6cddfd255388b58c61e25b091047f6814183e1d63741c8df8dcd65a2/virtualenv-16.1.0-py2.py3-none-any.whl (1.9MB)
Collecting virtualenv-clone>=0.2.5 (from pipenv)
  Downloading https://files.pythonhosted.org/packages/16/9d/6419a4f0fe4350db7fdc01e9d22e949779b6f2d2650e4884aa8aededc5ae/virtualenv_clone-0.4.0-py2.py3-none-any.whl
Requirement already satisfied: pip>=9.0.1 in /usr/local/lib/python3.6/site-packages (from pipenv) (18.0)
Installing collected packages: certifi, virtualenv, virtualenv-clone, pipenv
Successfully installed certifi-2018.11.29 pipenv-2018.11.26 virtualenv-16.1.0 virtualenv-clone-0.4.0
You are using pip version 18.0, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
+ yarn
yarn install v1.12.3
warning package.json: No license field
warning No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning " > bootstrap@4.1.3" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.1.3" has unmet peer dependency "popper.js@^1.14.3".
[4/4] Building fresh packages...
Done in 0.76s.
+ cp -p .env.example .env
+ pipenv install --system --deploy
Installing dependencies from Pipfile.lock (bc4449)…
+ pipenv run django-admin migrate
Creating a virtualenv for this project…
Pipfile: /buza-website/Pipfile
Using /usr/local/bin/python (3.6.6) to create virtualenv…
⠙ Creating virtual environment...Already using interpreter /usr/local/bin/python
Using base prefix '/usr/local'
New python executable in /root/.local/share/virtualenvs/buza-website-tHm4IlVX/bin/python
Installing setuptools, pip, wheel...
done.

✔ Successfully created virtual environment! 
Virtualenv location: /root/.local/share/virtualenvs/buza-website-tHm4IlVX
Loading .env environment variables…
Operations to perform:
  Apply all migrations: admin, auth, buza, contenttypes, sessions, taggit
Running migrations:
  Applying buza.0004_remove_question_topics... OK
  Applying buza.0005_subject_short_title... OK
Removing intermediate container da9e5e261e73
 ---> a4feb1c617db
Step 9/9 : EXPOSE 8000
 ---> Running in 3a8308eeb3ab
Removing intermediate container 3a8308eeb3ab
 ---> 886276a30aa5
Successfully built 886276a30aa5
</details>

zoidyzoidzoid committed:
Make .dockerignore work

I'm not a huge fan of this, especially the change to settings_env_dev,
but this is prettier than my previous attempt:
https://github.com/buza-project/buza-website/commit/a13faacc52fd81b07fe274b5853177362a6e0cab

sewagodimo commented:
yeahy, thank you for the PR :purple_heart: 
Good to merge, it passed on dockerhub

PiDelport commented on setup.py:
```diff
@@ -8,6 +9,15 @@
+    print("Disabling setuptools-scm")
+    extra_kwargs['use_scm_version'] = False
+
+
```
Sorry, late review!

This and the related changes to disable setuptools-scm will actually cause problems, because the manifest will no longer be generated correctly. It should be better to keep setuptools-scm always enabled, and rather install git inside the dev container, so that everything works together.

I'll create a follow-up PR with these changes.

PiDelport commented on src/buza/settings_env_dev.py:
```diff
@@ -11,7 +11,10 @@
+if 'IN_DOCKER' not in os.environ:
+    assert checkout_dir.joinpath('.git').exists(), checkout_dir
+else:
+    assert checkout_dir.exists(), checkout_dir
```
(as per above, regarding having git inside the dev container)

PiDelport reviewed (commented):
Add notes about setuptools-scm and git inside the container.

PiDelport commented:
Follow-up PR: #78
